src/Umbraco.Core/Services/LocalizedTextServiceFileSources.cs

```diff
@@ -179,7 +180,18 @@ private IEnumerable<IFileInfo> GetLanguageFiles()
         {
             result.AddRange(
                 new PhysicalDirectoryContents(_fileSourceFolder.FullName)
-                    .Where(x => !x.IsDirectory && x.Name.EndsWith(".xml")));
+                    .Where(x => !x.IsDirectory && !x.Name.Contains("user") && x.Name.EndsWith(".xml"))); // Filter out *.user.xml
+        }
+
+        if (_supplementFileSources is not null)
+        {
+            // Get only the .xml files and filter out the user defined language files (*.user.xml) that overwrite the default
+            var newLangFiles = _supplementFileSources.Where(x => !x.FileInfo.Name.Contains("user") && x.FileInfo.Name.EndsWith(".xml"));
+
+            foreach (var item in newLangFiles)
+            {
+                result.Add(item.FileInfo);
+            }
         }
```

Comment:
```suggestion
        if (_supplementFileSources is not null)
        {
            // Get only the .xml files and filter out the user defined language files (*.user.xml) that overwrite the default
            result.AddRange(_supplementFileSources
                .Where(x => !x.FileInfo.Name.Contains("user") && x.FileInfo.Name.EndsWith(".xml"))
                .Select(x => x.FileInfo));
        }
```

Very minor thing, but we can use Linq here and follow the same pattern as above and below this code block 😄